debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java

```diff
@@ -654,12 +654,22 @@ protected void handleCommit(OraclePartition partition, LogMinerEventRow row) thr
             return;
         }
 
-        counters.commitCount++;
-
         int numEvents = (transaction == null) ? 0 : getTransactionEventCount(transaction);
         LOGGER.debug("Committing transaction {} with {} events (scn: {}, oldest buffer scn: {}): {}",
                 transactionId, numEvents, row.getScn(), smallestScn, row);
 
+        // There are situations where Oracle records empty transactions in the redo and these
+        // do not make any changes. In such cases, LogMiner fails to reconcile the thread id
+        // for the commit, and leaves it assigned to 0. This ultimately leads to the commit
+        // recorded for the wrong redo thread. Given that we cannot just "guess" the thread
+        // as there may not have been a transaction recorded at all when LOB isn't enabled,
+        // all we can do is ignore the COMMIT.
+        if (row.getThread() == 0 && numEvents == 0) {
```

Comment:
What about adding a debug or trace log that we are going to skip this transaction?

Reply:
Can certainly add a TRACE log entry, that's not an issue.

Reply:
yeah, possibly adding thread number into previous debug log messages would be the best (number of the log messages won't increase and the information that this tx will be skipped would be there as the thread number and number of events will be in that log message)

Reply:
I've updated the log message @vjuranek, LMK if that looks better to you.